imap_data_access/file_validation.py

```diff
@@ -400,6 +425,80 @@ def is_valid_repointing(input_repointing: str) -> bool:
 class SPICEFilePath(ImapFilePath):
     """Class for building and validating filepaths for SPICE files."""
 
+    # Covers:
+    # Historical Attitude (type: ah.bc)
+    # Predicted Attitude (type: ap.bc)
+    # Spin Files (type: spin.csv)
+    # Repoint Files (type: repoint.csv)
+    attitude_file_pattern = (
+        r"(?P<{}>imap)_"
+        r"(?P<{}>[\d]{{4}})_"
+        r"(?P<{}>[\d]{{3}})_"
+        r"(?P<{}>[\d]{{4}})_"
+        r"(?P<{}>[\d]{{3}})_"
+        r"(?P<{}>[a-zA-Z0-9\-_]+)\."
+        r"(?P<{}>ah.bc|ap.bc|spin.csv|repoint.csv)"
+    ).format(
+        "mission", "year_start", "doy_start", "year_end", "doy_end", "version", "type"
+    )
+
+    # Covers:
+    # Reconstructed (type: recon)
+    # Nominal (type: nom)
+    # Predict (type: pred)
+    # 90 Day Predict (type: 90days)
+    # Long Term Predict (type: long)
+    # Launch Predict (type: launch)
+    spacecraft_ephemeris_file_pattern = (
+        r"(?P<{}>imap)_"
+        r"(?P<{}>[a-zA-Z0-9\-]+)_"
+        r"(?P<{}>[\d]{{8}})_"
+        r"(?P<{}>[\d]{{8}})"
+        r"(?:|_v(?P<{}>[\d]*))\."
+        r"(?P<{}>bsp)"
+    ).format("mission", "type", "date_start", "date_end", "version", "extension")
+
+    # Covers:
+    # Planetary Ephemeris (type: "de")
+    # Planetary Constants (type: "pck")
+    # Leapsecond kernel (type: "naif")
+    # Spacecraft clock kernel (type: "imapsclk_")
+    spice_prod_ver_pattern = (
+        r"(?P<{}>[a-zA-Z\-_]+)" r"(?P<{}>[\d]+)\." r"(?P<{}>tls|tpc|bsp|tsc)"
+    ).format("type", "version", "extension")
+
+    # Covers:
+    # Frame: (type: 'tf')
+    spice_frame_pattern = (r"(?P<{}>imap)_" r"(?P<{}>[\d]+)\." r"(?P<{}>tf)").format(
+        "mission", "version", "type"
+    )
+
+    # Covers:
+    # Thruster files (type: sff)
+    sff_filename_pattern = (
+        r"(?P<{}>imap)_"
+        r"(?P<{}>[\d]{{4}})_"
+        r"(?P<{}>[\d]{{3}})_"
+        r"(?P<{}>[a-zA-Z0-9\-_]+)_"
+        r"(?P<{}>[\d]{{2}})\."
+        r"(?P<{}>sff)"
+    ).format("mission", "year", "doy_start", "mode", "version", "type")
+
+    # Covers:
+    # Metakernels (type: 'tm')
+    mk_filename_pattern = (
+        r"(?P<{}>imap)_" r"(?P<{}>[\d]{{4}})_" r"v(?P<{}>[\d]{{3}})\." r"(?P<{}>tm)"
+    ).format("mission", "year", "version", "type")
+
+    valid_spice_regexes = (
+        re.compile(attitude_file_pattern),
+        re.compile(spacecraft_ephemeris_file_pattern),
+        re.compile(spice_prod_ver_pattern),
+        re.compile(spice_frame_pattern),
+        re.compile(sff_filename_pattern),
+        re.compile(mk_filename_pattern),
+    )
```

Comment:
Do you want to combine them into a single match? I think you can do this as a large regex with the pipe operator.
```suggestion
    valid_spice_regexes = re.compile("|".join([attitude_file_pattern, spacecraft_ephemeris_file_pattern, spice_prod_ver_pattern, spice_frame_pattern, sff_filename_pattern, mk_filename_pattern)
```

Reply:
Yup!

Reply:
Actually it appears that the problem with a single match is that it complains that I have too many fields named "mission", "version", "type", etc, once  they all get combined into one big regex

Reply:
Ahh, that makes sense. Thanks for checking!